src/io/export_3mf_v2.cc

```diff
@@ -121,6 +121,26 @@ static bool append_polyset(std::shared_ptr<const PolySet> ps, Lib3MF::PWrapper&
       }
     }
 
+    Lib3MF::PBaseMaterialGroup pBaseMaterial = model->AddBaseMaterialGroup();
+
+    std::vector<Lib3MF_uint32> ids;
+    for(int i=0;i<out_ps->colors.size();i++) {
+      sLib3MFColor col;
+      std::string colname="Color" + i;
```

Comment:
That does not work as it looks like as it does pointer arithmetic before assignment.

int i = 3; std::string colname="Color" + i;
Result of that is colname = "or"!

Reply:
good catch. It works well up to maximum of 6 colors then segfaults ...